presto-main/src/main/java/io/prestosql/operator/scalar/BitwiseFunctions.java

```diff
@@ -75,4 +75,115 @@ public static long bitwiseXor(@SqlType(StandardTypes.BIGINT) long left, @SqlType
     {
         return left ^ right;
     }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.TINYINT)
+    public static long bitwiseLeftShiftTinyint(@SqlType(StandardTypes.TINYINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        return bitwiseLeftShift(num, shift);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.SMALLINT)
+    public static long bitwiseLeftShiftSmallint(@SqlType(StandardTypes.SMALLINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        return bitwiseLeftShift(num, shift);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.INTEGER)
+    public static long bitwiseLeftShiftInteger(@SqlType(StandardTypes.INTEGER) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        return bitwiseLeftShift(num, shift);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseLeftShiftBigint(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        return bitwiseLeftShift(num, shift);
+    }
+
+    private static long bitwiseLeftShift(long num, long shift)
```

Comment:
Abstracting this in a method is overkill. Inline the expression in all the methods that use it.